threatgrid/ctia
# Remove http routes for disabled entities

> Close #1016
> Related https://github.com/threatgrid/response/issues/389

Remove http routes and hide Swagger contexts for disabled entities.

for testing locally: 
- add `ctia.features.disable=asset,actor,sighting` to `resources/ctia-default.properties`
- start the server, open Swagger console
- check that contexts for Asset, Actor and Sighting are not shown
 





<a name="qa">[§](#qa)</a> QA
============================

1. Check property configuration for `ctia.features.disabled` record 
2. Run the app for different environments
3. Make sure that the routes and contexts for entities like Asset, AssetMapping, AssetProperties, TargetRecord are enabled everywhere, except the environments where they set to be disabled. e.g.: prod. 






<a name="release-notes">[§](#release-notes)</a> Release Notes
=============================================================


`intern:`
Related configuration changes: https://github.com/threatgrid/tenzin-config/pull/325



<a name="squashed-commits">[§](#squashed-commits)</a> Squashed Commits
======================================================================

0feab531 PR feedback
4b48e512 convolute sexps
849136f2 PR feedback: replace if/else with cond->
8aff4aa4 PR feedback: inline fixtures
7c2fbbc0 PR feedback: remove nils
1ed45c28 PR feedback: leave only one method to reduce ambiguity
4015d267 PR feedback: contrain schema
210a897d PR feedback: rename fn
a976ff13 api tags sorted alphabetically
914fea56 test: featuresService
975d27e5 remove routes for disabled entities
46e43c8e adds tests and docstrings
72892cb7 adds FeaturesService
